helix-term/src/ui/mod.rs

```diff
@@ -177,6 +177,9 @@ pub fn file_picker(root: PathBuf, config: &helix_view::editor::Config) -> Picker
         .max_depth(config.file_picker.max_depth)
         .filter_entry(move |entry| filter_picker_entry(entry, &absolute_root, dedup_symlinks));
 
+    walk_builder.add_custom_ignore_filename(".helix/ignore");
+    walk_builder.add_custom_ignore_filename(helix_loader::config_dir().join(".ignore"));
```

Comment:
The docs for [add_custom_ignore_filename](https://docs.rs/ignore/latest/ignore/struct.WalkBuilder.html#method.add_custom_ignore_filename) mention this takes precedence, which I'd argue doesn't make sense for a global ignore file and that's why I have opted for [add_ignore](https://docs.rs/ignore/latest/ignore/struct.WalkBuilder.html#method.add_ignore) in #8300 .

Reply:
I actually disagree. The ignore file is more specficic because it's specific to helix and should take priority. 

By making the global helic specific ignore file take priority that would allow people to always show some folder (like `node_modules`) in helix but still add it to gitignore.

I think the case where somebody whitelists a glob in their global helix ignorefile and still wants to ignore that file for a specific project is pretty niche and can be handled by adding that file to the helix specific ignore file for that project.


Priority should be 

* local helix ignore file 
* global high ignore file
* all other ignore files

This is what using `add_custom_ignorefile` accomplishes. The problem here was just the order (you should add the global ignore file first and then the local one so the local one has higher priority)